### PR TITLE
[VIDEO-1923] Add Linux support

### PR DESCRIPTION
**Pull request by btgoodwin**
## Description

Updated the github action to support Ubuntu 20.04 installations from source and leveraged caching to help speed up execution.

**Ticket(s) Addressed:**

* VIDEO-1923

## Design Summary

* Added a map of distro configurations, min/max semver, etc. to help define a few useful configurations that can easily be fed into `@actions/exec`.
* Added rudimentary Linux OS distribution and version checking by parsing `/etc/*-release` for `NAME` and `VERSION_ID`.  It's not perfect but there is no codified way to do this, so this is probably "pretty close to good."
* Extended action definition to include `repoUrl` in case we need to change source targets in this configuration
* `version` is used as the tag to pull from source
* Implemented `@actions/cache` to save off the gstreamer source directory between runs under a key involving the version, platform, distro, and so forth to uniquely identify the caches.
* Updated readme accordingly.
* Added cache version so we can invalidate previous caches if we need to change the linux procedure.

**Configuration changes:**

* `repoUrl` defaults to the published gstreamer source gitlab.

## Testing Notes and Procedure

Testing is automated; no procedure at this time.

The unit test for `platform === linux` has been disabled b/c in that environment `@actions/cache` and the lack of its environment variables/services is causing the test to crash even if I guard against cache calls by not calling them with `isFeatureAvailable()`.  The solutions described on other forums don't sound like a good way forward for us at this time, and the other automated test where we actually run the action in a matrix with the other platforms does succeed, so disabling this test is okay as it does the same thing but from a different environment.

## Related Pull Requests

None

### CODE REVIEW

File: index.test.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ test('test runs', () => {
...
+    // we cannot run this in the test environment because the ACTIONS_ variables
+    // are not set and no service is present.  Trying to avoid using the cache
+    // API by checking isFeatureAvailable does not prevent the code from crashing
+    // when trying to check those variables.
+    console.log(`skipping test on ${process.platform}`);
+  }
+  else {
+    let result = cp.execSync(`node ${ip}`, {env: process.env}).toString();
+    console.log(result);
+  }
```
Comment (jawilson) [inline index.test.js:21]: The `jest` testing framework actually has built-in skipping functionality, but not conditionally. I would recommend doing something like [this](https://stackoverflow.com/a/69396619/5321669) to skip the test entirely (doing the platform check to decide to call `.skip()`) instead of just logging in the code. It'll correctly report that the test was skipped.

**Comment by btgoodwin** [comment]
I say punt. I updated the other ticket ([for implementing caching](https://blinemedical.atlassian.net/browse/VIDEO-1924)) to suggest we either need to drop the test, which is actually a duplicate, or come up with a way to work around it / mock it so all supported platforms can run the test (and any others we come up with).
